osp/core/utils/general.py

```diff
@@ -55,13 +56,19 @@ def get_rdf_graph(session=None):
     Returns:
         rdflib.Graph: The resulting rdf Graph
     """
+    if session is not None:
+        if not isinstance(session, Session):
+            raise TypeError(
+                f"Invalid argument: {session}."
+                f"Function can only be called on (sub)classes of {Session}."""
+            )
     from osp.core.cuds import Cuds
-    from osp.core import ONTOLOGY_NAMESPACE_REGISTRY
+    from osp.core.namespaces import _namespace_registry
     session = session or Cuds._session
     graph = rdflib.Graph()
     for triple in session.get_triples():
         graph.add(triple)
-    for namespace in ONTOLOGY_NAMESPACE_REGISTRY:
+    for namespace in _namespace_registry:
```

Comment:
Since we already have the graph of the namespace_registry, this looks unnecessarily inefficient.
I guess we should fiy this as well.

What do you think @aaronAB1993

Reply:
I mean that we iterate over all entities and call get_triples

Reply:
So what you suggest is to merge the namespace registry graph with the "CUDS graph" and then return?

Reply:
Exactly, although I'm not sure how to `merge two graphs`  ideally.
I've read in the rdflib docs, that using the python union operator can cause problems.

Reply:
Yes but only for blank nodes, right? Do we expect any blank nodes for the CUDS graph?

Reply:
Hmm you are right. There shouldn't be any in the CUDS graph.

Reply:
Then this should actually be pretty straighforward, right?

Reply:
I fixed this in the latest commit. It is better now.